daemon/daemon.go

```diff
@@ -481,6 +493,10 @@ func (d *Daemon) HandleRestart(t state.RestartType) {
 	// die when asked to restart (systemd should get us back up!) etc
 	switch t {
 	case state.RestartDaemon:
+		d.mu.Lock()
```

Comment:
(unreleated to this PR) I wonder if we should simply always grab the d.mu.Lock() in HandleRestart